tests/contrib/aiohttp/test_aiohttp_client.py

```diff
@@ -0,0 +1,177 @@
+import os
+
+import aiohttp
+import pytest
+
+from ddtrace.contrib.aiohttp import patch
+from ddtrace.contrib.aiohttp import unpatch
+from tests.utils import override_http_config
+
+from ..config import HTTPBIN_CONFIG
+
+
+HOST = HTTPBIN_CONFIG["host"]
+PORT = HTTPBIN_CONFIG["port"]
+SOCKET = "{}:{}".format(HOST, PORT)
+URL_200 = "http://{}/status/200".format(SOCKET)
+URL_500 = "http://{}/status/500".format(SOCKET)
+
+
+@pytest.fixture(autouse=True)
+def patch_aiohttp_client():
+    patch()
+    yield
+    unpatch()
+
+
+@pytest.mark.asyncio
+async def test_200_request(snapshot_context):
+    with snapshot_context():
+        async with aiohttp.ClientSession() as session:
+            async with session.get(URL_200) as resp:
+                assert resp.status == 200
+
+
+@pytest.mark.asyncio
+async def test_500_request(snapshot_context):
+    with snapshot_context():
+        async with aiohttp.ClientSession() as session:
+            async with session.get(URL_500) as resp:
+                assert resp.status == 500
+
+
+@pytest.mark.asyncio
+async def test_200_request_distributed_tracing(tracer, test_spans):
+    class MockRequest(object):
+        headers = None
+
+        def __init__(self, *args, **kwargs):
+            MockRequest.headers = kwargs.get("headers")
+
+    async with aiohttp.ClientSession() as session:
+        session._request_class = MockRequest
+        try:
+            async with session.get(URL_200) as resp:
+                assert resp.status == 200
+        except AttributeError:
+            # Expected because MockRequest doesn't have the required attributes
+            pass
+
+    for h in ("x-datadog-trace-id", "x-datadog-parent-id", "x-datadog-sampling-priority"):
+        assert h in MockRequest.headers
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("variant", ["pin", "global"])
+async def test_distributed_tracing_disabled(ddtrace_run_python_code_in_subprocess, variant):
+    code = f"""
+import asyncio
+import sys
+import aiohttp
+from ddtrace import Pin
+from tests.contrib.aiohttp.test_aiohttp_client import URL_200
+
+class MockRequest(object):
```

Comment:
why not just ping `httpbin` and assert the headers sent from that?

Reply:
I suppose it really only saves you this mock, otherwise the test is basically the same